app/views/spree/users/_social.html.erb

```diff
@@ -1,28 +1,30 @@
 <% if Spree::AuthenticationMethod.active_authentication_methods? %>
-<% @body_id = 'login' %>
-<div id="existing-customer">
-  <% if spree_current_user.user_authentications %>
-    <% unless spree_current_user.user_authentications.empty? %>
-      <p><strong><%= Spree.t(:you_have_signed_in_with_these_services) %>:</strong></p>
-      <div class="authentications">
-        <% for user_authentication in spree_current_user.user_authentications %>
-          <div class="authentication">
-            <div class="provider columns two">
-              <%= content_tag(:i, '', class: "icon-spree-#{user_authentication.provider.to_url}-circled columns") %>
-              <%= user_authentication.provider %>
+  <% @body_id = 'login' %>
+  <div id="existing-customer">
+    <% if spree_current_user.user_authentications %>
+      <% unless spree_current_user.user_authentications.empty? %>
+        <p><strong><%= Spree.t(:you_have_signed_in_with_these_services) %>:</strong></p>
+        <div class="authentications">
+          <% for user_authentication in spree_current_user.user_authentications %>
+            <div class="authentication">
+              <div class="provider columns two">
+                <%= content_tag(:i, '', class: "icon-spree-#{user_authentication.provider.to_url}-circled columns") %>
+                <%= user_authentication.provider %>
+              </div>
+              <div class="uid columns two"><%= user_authentication.uid %></div>
+              <%= link_to 'X', user_authentication, data: { confirm: "#{Spree.t(:remove_authentication_option_confirmation)}" }, method: :delete, class: 'remove' %>
             </div>
-            <div class="uid columns two"><%= user_authentication.uid %></div>
-            <%= link_to 'X', user_authentication, data: { confirm: "#{Spree.t(:remove_authentication_option_confirmation)}" }, method: :delete, class: 'remove' %>
-          </div>
-        <% end %>
-        <div class="clear"></div>
-      </div>
+          <% end %>
+          <div class="clear"></div>
+        </div>
+      <% end %>
     <% end %>
+    
+    <% if Spree::AuthenticationMethod.available_for(spree_current_user).exists? %>
```

Comment:
I'm fairly sure `.exists?` will run an additional query that won't hit the query cache when `.available_for` is used inside the partial. `.present?` or something would be preferable.